doc/iterations/2024.5/shapes/configurable_visibility.adoc

```diff
@@ -0,0 +1,47 @@
+= Configurable visibility
+
+== Problem
+
+Sirius Web doesn't provide a way for specifiers to set the default visibility of a node (hidden/visible). 
+It also doesn't provide a way to let specifiers define tools that hide/reveal nodes as part of other operations (e.g. to reveal a hidden node when a child node is created).
+Nodes are always created visible, and the only way to hide them right now is that the end user does it manually.
+
+== Key Result
+
+- Specifiers must be able to define NodeDescriptions that are hidden/visible by default.
+- Specifiers must be able to call hide/reveal services in their tool to hide/reveal a set of nodes.
+
+== Solution
+
+A new field _Is Hidden By Default Expression_ should be displayed in NodeDescription. 
+The field contains an boolean expression which is evaluated on the first rendering of the node to set its initial hidden/visible state.
+If the expression is empty it is evaluated to `false`, which means that the node is visible by default (this preserves Sirius Web's default behavior to show new nodes by default). 
+
+The `diagramServices` variable should contain new services to hide/reveal nodes.
+
+== Scenario
+
+1. Default Visibility State
+- A specifier adds a NodeDescription in the view DSL
+- The _Is Hidden By Default Expression_ field is available and its default value is empty, which means visible
+- The specifier can define an AQL expression to that returns a boolean value to set the default visibility of the node (`true` for hidden, `false` for visible)
+- When an end user creates a node represented in this manner in a diagram, the node is hidden/visible according to this expression
+- If the user manually hides/reveals the node the value returned by the expression is not taken into account (the expression can't override user-defined state).
+
+2. Hide/Reveal operation
+- A specifier creates a tool in the view DSL
+- The specifier creates an operation, for example ChangeContext
+- The specifier fills the expression of its operation with a call to the hide/reveal service, for example `diagramServices.hide(nodeId)`
```

Comment:
I would add fade too

Reply:
Done, I propose 2 duos of services:
- hide/reveal that only deals with the hidden aspect
- fade/unfade that only deals with the faded aspect

Let me know if the names don't work for you. There might be a confusion for reveal, which is used in Sirius Web for both faded and hidden elements.
![image](https://github.com/eclipse-sirius/sirius-web/assets/6516281/96cb9faf-4571-4ec9-aa7c-af131bafde90)

Reply:
Well I think there should be three services since there are only three states and not two different states that can be toggled. A node can be visible, hidden or faded. Unfade and reveal would do the same thing which is set the node back to the normal state. By having two ways to do the same thing, the specifier would have to figure out if the node is current hidden or faded to find out which service to use.

Something like this would be easier to use:

- hide(...)
- fade(...)
- show(...)

`show` would restore back a diagram element to its normal state regardless of whether it was faded or hidden because it does not make any difference.

Reply:
As you said, a `ViewModifier` has only 3 modes, but a node can have multiple `ViewModifiers`. We can fade a node then hide it, and if we reveal it via the "reveal hidden elements" tool on the diagram it will make it visible, but still faded. This wouldn't be possible with a single `show(...)` method.

![FadeAndHide](https://github.com/eclipse-sirius/sirius-web/assets/6516281/ca226059-84f8-437b-880a-378393237030)

Reply:
> As you said, a ViewModifier has only 3 modes, but a node can have multiple ViewModifiers.

Yes indeed, I forgot about that part -.-"

I'm not sure that a set of view modifiers was the best to do that but that's the API we have so yes, your proposal with two sets of operations is better, let's go for that. I'll have one last look at the shape and merge it if there are no other issues